packages/ui/libs/ui/utils/src/permissions/hooks/usePermissions.ts

```diff
@@ -1,50 +1,58 @@
-import {
-  IUser,
-  OrganizationStatus,
-  UserStatus,
-} from '@energyweb/origin-backend-core';
+import { useTranslation } from 'react-i18next';
+import { OrganizationStatus, UserStatus } from '@energyweb/origin-backend-core';
+import { useUserControllerMe } from '@energyweb/origin-backend-react-query-client';
+import { useAccountControllerGetAccount } from '@energyweb/exchange-react-query-client';
 import { defaultRequirementList } from '../defaultRequirementList';
-import { IPermission, IPermissionRule, Requirement } from './../../types';
+import {
+  IPermissionRule,
+  IPermissionReturnType,
+  Requirement,
+} from './../../types';
+
+export const usePermissions = (
+  config = defaultRequirementList
+): IPermissionReturnType => {
+  const { t } = useTranslation();
+
+  const { data: userData, isLoading: userLoading } = useUserControllerMe();
+  const { data: account, isLoading: exchangeAddressLoading } =
+    useAccountControllerGetAccount();
```

Comment:
origin-ui-utils is a generic package for helpers functions. So we cant store the api calls here, because it will make generic component dependent on the exact implementation of exchange-api and origin-backend. 
You should keep here only the main logic for checking the if the requirements are met. 
All other custom logic should be implemented in a dedicated package.